File: seed/ruby-sdk-v2/examples/lib/node.rb
# frozen_string_literal: true

module Types
    module Types
        class Node < Internal::Types::Model
            field :name, String, optional: true, nullable: true
            field :nodes, Array, optional: true, nullable: true
            field :trees, Array, optional: true, nullable: true
        end
    end
end

File: seed/pydantic-v2/examples/baseline/resources/types/types/node.py
from pydantic import BaseModel
from typing import Optional, List
from resources.types.types.node import Node
from resources.types.types.tree import Tree
from dt import datetime
from core.datetime_utils import serialize_datetime


class Node(BaseModel):
    name: str
    nodes: Optional[List[Node]] = None
    trees: Optional[List[Tree]] = None

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}
